### PR TITLE
Add game metadata on Goban Div

### LINKED ISSUE #1392: Provide game metadata in the Goban div
People would like to [customize their experience based on the context of the game they are playing.](https://forums.online-go.com/t/features-everyone-secretly-want-on-ogs-but-will-never-be-implemented/32125/63?u=eugene)

**Describe the solution you'd like**

They want metadata in css classes:

```
<div class="Goban" data-game-id="123" data-tournament-id="1234">
<div class="Goban" data-game-id="123" data-ladder-id="1234">
```

### PR DESCRIPTION
**Pull request by GreenAsJade**
Fixes #1392

## Proposed Changes

Make PersistentElement take a hash of extra properties to put on the element, and put the game metadata on the Goban in this way.

### CODE REVIEW

File: src/views/Game/Game.styl [hunk 1]
```diff
@@ -397,6 +397,7 @@ goban-view-bar-width=400px
         align-items: stretch;
         width: 100%;
 
+        // Why? ...
```
Comment (benjaminpjones) [inline src/views/Game/Game.styl:400]: Haha [I just delete these when I see them](https://github.com/online-go/online-go.com/pull/1383/commits/e395201c3d24bd7277895343ff3dc55a78525577)... if there's a reason for them, I'd like to know as well!

**Comment by pdg137** [comment]
Here's the HTML I get from this, on a *game* page:

```html
<div class="Goban" game-id="10021" ladder-id="10">
  <div class="Goban" style="width: 609px; height: 609px; background-color: rgb(220, 179, 92); background-image: url(&quot;https://cdn.online-go.com/5.1/img/kaya.jpg&quot;); top: 197.5px; left: 8.5px;">
  ...
```

The info is on the outer Goban div instead of the one inside of it that seems to be the more important one. Is the outer one even necessary at all?

Also, can you get this into Gobans within a "GameList"? I'm hoping this info will be useful for distinguishing games within lists.

**Comment by GreenAsJade** [comment]
@pdg137 It's pretty ugly that we have a Goban inside a Goban isn't it?   But it would be a braver person than me to attempt to flatten that - goodness knows what css would break.   

Given that you know that the structure is like this, do you care?  You get the info you need from the outside one, you set what you need to on the inside one...

It's a heck of a lot cleaner to add it on the outside one, but I can poke around on adding it onto the inside one if "I have to" 😁

**Comment by GreenAsJade** [comment]
... and it will be a lot easier to add it to the game list as well, if it is on the outer thing.  

Though regrettably the class of the outer thing is "small board" with a space.  I'm not sure what the "small" does - maybe font stuff, but the board is used.

We could add "Mini-goban", so it becomes "Mini-goban small board".

Either that or we have to do lots of yucky stuff to put those properties on the Goban inside, in both Game and GameList...

**Comment by GreenAsJade** [comment]
@anoek this should get "work in progress" on it.  Not sure how to do that.

I'm starting to think that really the attributes should be on the real Goban div...

**Comment by pdg137** [comment]
I can probably handle having them on the outer div, but then I'm worried about this change legitimizing that element.  Right now the outer "Goban" one doesn't seem to be doing anything, and it's going to be harder to remove it once it has some useful properties.

The "small board" one seems to be where the green highlighting is applied when it's your move, not sure whether it does anything else.

**Comment by GreenAsJade** [comment]
@pdg137  For the record, I tried collapsing out the outer Goban, as feared it breaks the CSS, so that job is non-trivial.

**Comment by GreenAsJade** [comment]
@pdg137  @anoek  I think this is what was requested, now, for the Goban.

MiniGoban suport for the same thing, in GameList will, require a back end change to provide the ladder and tournament id with each game in the game list, so that's better in a separate request.   Maybe let's see if this takes hold first.

File: src/views/Game/Game.styl (hunk 1 above)
Comment (anoek) [reply src/views/Game/Game.styl:400]: Yeah those are always safe to delete, that's just slop - looks like that came in during the initial import into the open source repo 4 years ago - slop.

**Comment by anoek** [comment]
Sounds good, looks good, I'll do some testing and merge it soon.

I *think* it should be possible to remove that outer Goban, all of that was implemented before we were comfortable using `flexbox` so that can likely be gotten rid of, although I think we still need the `Goban.inner-container` to deal with layout of the text around the board, and providing a `relative` environment for the canvas div container. But yeah, unless there's a particular reason to go down that rabbit hole, it's almost certainly not worth it.

File: src/views/Game/Game.styl (hunk 1 above)
Comment (BHydden) [reply src/views/Game/Game.styl:400]: We still love you anoek-senpai ❤ thanks for your hard work!

**Comment by GreenAsJade** [comment]
@pdg137  Over to you ;)

**Comment by pdg137** [comment]
@GreenAsJade okay [here you go](https://forums.online-go.com/t/how-to-customize-boards-for-each-tournament-ladder-game/34459)!

**Pull request merged**